dynamic-simulation/dynamic-simulation-api/src/main/java/com/powsybl/dynamicsimulation/DynamicSimulationResultImpl.java

```diff
@@ -7,37 +7,47 @@
 package com.powsybl.dynamicsimulation;
 
 import java.util.Collections;
+import java.util.List;
 import java.util.Map;
 import java.util.Objects;
 
-import com.powsybl.timeseries.StringTimeSeries;
 import com.powsybl.timeseries.TimeSeries;
 
 /**
  * @author Marcos de Miguel {@literal <demiguelm at aia.es>}
+ * @author Laurent Issertial {@literal <laurent.issertial at rte-france.com>}
  */
 public class DynamicSimulationResultImpl implements DynamicSimulationResult {
 
-    private final boolean status;
-    private final String logs;
+    private final Status status;
+    private final String statusText;
     private final Map<String, TimeSeries> curves;
-    private final StringTimeSeries timeLine;
+    private final List<TimelineEvent> timeLine;
 
-    public DynamicSimulationResultImpl(boolean status, String logs, Map<String, TimeSeries> curves, StringTimeSeries timeLine) {
-        this.status = status;
-        this.logs = logs;
+    public DynamicSimulationResultImpl(Status status, String statusText, Map<String, TimeSeries> curves, List<TimelineEvent> timeLine) {
+        this.status = Objects.requireNonNull(status);
+        this.statusText = Objects.requireNonNull(statusText);
         this.curves = Objects.requireNonNull(curves);
         this.timeLine = Objects.requireNonNull(timeLine);
+        timeLine.forEach(Objects::requireNonNull);
+    }
+
+    public DynamicSimulationResultImpl(Status status, Map<String, TimeSeries> curves, List<TimelineEvent> timeLine) {
+        this(status, "", curves, timeLine);
+    }
+
+    public static DynamicSimulationResultImpl createSucceededResult(Map<String, TimeSeries> curves, List<TimelineEvent> timeLine) {
```

Comment:
Now that succeed has been renamed to success, we should probably rename this into `createSuccessResult`.